### PR TITLE
[JW8-10570] Sync with upstream feature/v1.0.0

### PR DESCRIPTION
**Pull request by robwalch**
### This PR will...
Sync LHLS branch with upstream v1.0.0 PR.

We have PR open upstream with branch feature/v1.0.0 against master:
https://github.com/video-dev/hls.js/pull/2370

### Why is this Pull Request needed?

This is a working branch but more stable than LHLS. We should stay in sync with upstream for visibility (on both ends) into status of changes and issues.

The upstream work includes:
- 0.13 changes and fixes from upstream
- bug fix for live stream sync
- lint and type-check fixes
- unit test fixes
- use npm instead of yarn

### Are there any points in the code the reviewer needs to double check?
The merge branch restores `"version"` in package.json which is not present upstream. yarn and npm errors when our build uses `yarn add` to modify the package.json to a commit where the package.json is technically invalid.

### Resolves issues:
JW8-10570

### CODE REVIEW

File: docs/design.md [hunk 1]
```diff
@@ -172,19 +172,19 @@ design idea is pretty simple :
...
-   - example dummy remuxer
+    - example dummy remuxer
   - [src/remux/mp4-generator.js][]
-   - in charge of generating MP4 boxes
-     - generate Init Segment (moov)
-     - generate samples Box (moof and mdat)
+     - in charge of generating MP4 boxes
+       - generate Init Segment (moov)
+       - generate samples Box (moof and mdat)
   - [src/remux/mp4-remuxer.js][]
```
Comment (zetagame) [inline docs/design.md]: Should we change this to mp4-remuxer.ts?

File: src/controller/timeline-controller.ts [hunk 2]
```diff
@@ -1,24 +1,30 @@
...
+  private tracks: Array<any> = [];
+  private initPTS: Array<number> = [];
+  private unparsedVttFrags: Array<{frag: Fragment, payload: any}> = [];
+  private cueRanges: { [trackName: string]: Array<any> } = {};
+  private captionsTracks: { [trackName: string]: any } = {};
+  private captionsProperties: any;
+  private cea608Parser?: Cea608Parser;
+  private lastSn: number = -1;
+  private prevCC: number = -1;
+  private vttCCs: any = null;
```
Comment (zetagame) [inline src/controller/timeline-controller.ts]: Does it make sense to specify types for vttCCs and Cues?

File: tests/unit/controller/buffer-controller.js [hunk 3]
```diff
@@ -16,6 +16,122 @@ describe('BufferController tests', function () {
...
+  //
+  //   it('flushes to a max of one targetDuration from currentTime, regardless of liveBackBufferLength', function () {
+  //     mockMedia.currentTime = 15;
+  //     bufferController._levelTargetDuration = 5;
+  //     hls.config.liveBackBufferLength = 0;
+  //     bufferController.flushLiveBackBuffer();
+  //     expect(removeStub).to.have.been.calledWith('video', mockSourceBuffer.video, 0, 10);
+  //   });
+  //
+  //   it('should trigger clean back buffer when there are no pending appends', function () {
```
Comment (zetagame) [inline tests/unit/controller/buffer-controller.js]: What are these for?

File: docs/design.md (hunk 1 above)
Comment (robwalch) [reply docs/design.md]: good catch

**Pull request merged**